site/content/how-to/traffic-management/response-headers.md

```diff
@@ -0,0 +1,209 @@
+
+### Send Traffic to the Headers Application
+
+We will use `curl` with the `-i` flag to access the application and include the response headers in the output:
+
+```shell
+curl -i --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/headers
+```
+
+```text
+HTTP/1.1 200 OK
+Server: nginx/1.25.5
+Date: Mon, 06 May 2024 19:08:39 GMT
+Content-Type: text/plain
+Content-Length: 2
+Connection: keep-alive
+X-Header-Unmodified: unmodified
+X-Header-Add: add-to
+X-Header-Set: overwrite
+X-Header-Remove: remove
+
+ok
+```
+
+In the output above, you can see that the headers application adds the following custom headers to the response:
+
+- X-Header-Unmodified: unmodified
+- X-Header-Add: add-to
+- X-Header-Set: overwrite
+- X-Header-Remove: remove
+
+In the next section we will modify these headers by adding a ResponseHeaderModifier filter to the headers HTTPRoute.
+
+### Update the HTTPRoute to Modify the Response Headers
+
+Let's update the HTTPRoute by adding a `ResponseHeaderModifier` filter:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: headers
+spec:
+  parentRefs:
+  - name: gateway
+    sectionName: http
+  hostnames:
+  - "cafe.example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /headers
+    filters:
+    - type: ResponseHeaderModifier
+      responseHeaderModifier:
+        set:
+        - name: X-Header-Set
+          value: overwritten-value
+        add:
+        - name: X-Header-Add
+          value: this-is-the-appended-value
+        remove:
+        - X-Header-Remove
+    backendRefs:
+    - name: headers
+      port: 80
+EOF
+```
+
+Notice that this HTTPRoute has a `ResponseHeaderModifier` filter defined for the path prefix `/headers`. This filter:
+
+- Sets the value for the header `X-Header-Set` to `overwritten-value`.
+- Adds the value `this-is-the-appended-value` to the header `X-Header-Add`.
+- Removes `X-Header-Remove` header.
+
+
+### Send Traffic to the Modified Headers Application
+
+We will send a curl request to the modified `headers` application and verify the response headers are modified.
+
+```shell
+curl -i --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/headers
+```
+
+```text
+HTTP/1.1 200 OK
+Server: nginx/1.25.5
+Date: Mon, 06 May 2024 17:58:33 GMT
+Content-Type: text/plain
+Content-Length: 2
+Connection: keep-alive
+X-Header-Unmodified: unmodified
+X-Header-Add: add-to
+X-Header-Add: this-is-the-appended-value
+X-Header-Set: overwritten-value
+
+ok
+```
+
+In the output above, you can see that the headers application modifies the following custom headers:
+
+In the output above you can notice the modified response headers as the `X-Header-Remove` header is absent. The header `X-Header-Add` gets appended with the new value and `X-Header-Set` gets overwritten to `overwritten-value` as defined in the *HttpRoute*.
+
```

Comment:
Let's add one extra line here pointing out that the `X-Header-Unmodified` headers is unchanged because we did not include it in the filter.